.github/workflows/lint.yml

```diff
@@ -27,4 +29,19 @@ jobs:
           pip install -r data_requirements.txt
       - name: Run linter
         run: make lint
-        
\ No newline at end of file
+      - name: Get changed files
+        id: changed-files
+        run: |
+          if ${{ github.event_name == 'pull_request' }}; then
+            echo "changed_files=$(git diff --name-only -r HEAD^1 HEAD | xargs)" >> $GITHUB_OUTPUT
+          else
+            echo "changed_files=$(git diff --name-only ${{ github.event.before }} ${{ github.event.after }} | xargs)" >> $GITHUB_OUTPUT
+          fi
+      - name: Ensure Changelog Updated
+        run: |
+          for file in ${{ steps.changed-files.outputs.changed_files }}; do 
+            if [ "${file}" == CHANGELOG.md ]; then 
```

Comment:
I wonder if this is too strict. We currently do some hotfix PRs that fixes a couple of lines.

(I think in principle we shouldn't do that.) But not sure if the added friction here is necessary.

Reply:
I think the concern here is that we'd end up with placeholder changelog lines if we really enforce every PR modifies it.

Reply:
That's fair. I can remove the check, but as we just need to do a good job of policing the file at our discretion 👍

Maybe I'll add a note to to the PR template instead